### DISCUSSION TITLE
How to target :hover on the element being hovered and not the parents?

### DISCUSSION THREAD

**Discussion by maxpaleo** [tailwindlabs/tailwindcss#15059] [Help]
Does anyone know how to target `:hover` on the element that the cursor is currently on without triggering it for all its parents as well? 

Each element is wrapped in a div with a class that should turn the bg of the currently hovered element green, but it keeps targeting all its parents as well. 

I assume that this is because technically, the cursor is above these elements as well. 

Is there a way to avoid this? 

```tsx
/** 
 * This div wraps every element in the ui. 
 * - It should only turn the bg green on the element the cursos is actually on, not its parents. 
 */
<div tabIndex={0} className={cn("hover:bg-green-400")}>
  {children}
</div>
```

<img width="896" alt="Screenshot 2024-11-20 at 10 40 12 PM" src="https://github.com/user-attachments/assets/aac76f06-9b16-410a-84bc-1a84e281cdee">

**Comment by wongjn** [comment]
You could try `[&_:hover:not(:has(:hover))]:bg-green-400`: https://play.tailwindcss.com/9idDLsMugF
